### DISCUSSION TITLE
Optional Field in Loop

### DISCUSSION THREAD

**Discussion by hooksie1** [cue-lang/cue#1894] [Q&A] [answered]
We are generating openapi specs and have a list of response types that we would like to make optional. Adding the question mark after `"\(v)"` doesn't seem to work. For example:

```
#Series: {
    name: string
    data: string
}

_ResponseTypes: [
    "BytesSent",
    "BytesReceived",
    "BytesPending"
]

Response: {
    for v in _ResponseTypes {
        "\(v)"?: [...#Series]
    }
}
```

Maybe we are approaching this the wrong way, but it seemed like looping over the response types was the easiest since we have quite a number of them and they all hold the same type of data. Adding the question mark just doesn't do anything. It doesn't give any errors, just doesn't seem to do anything.

**Comment by verdverm** [comment]
What command are you running? What output are you seeing? What output are you expecting?

Generally it is helpful to provide working examples or reproducers: https://github.com/cue-lang/cue/wiki/Creating-test-or-performance-reproducers

https://cuelang.org/play/?id=TD7awYzT9hc#cue@export@cue has some output

**Comment by antong** [comment, marked as answer]
It seems it is not the loop that makes the fields not optional, but the string interpolation. I ran the following in the playground (https://cuelang.org/play/?id=u2ydPPGyZDA#cue@export@cue, v0.4.3):
```
#Series: {
    name: string
    data: string
}

Response: {
    a: [...#Series]
    b?: [...#Series]
    "c"?: [...#Series]
    let k="d"
    "\(k)"?: [...#Series]
}
// -> Response: {a: [], d: []}
```
As we can see, the quoted field "c" is made optional by the question mark, but not the one with the string interpolation (d). I do not know whether this is a feature or a bug. I'd guess a bug.

I'm sure there are lots of other ways to write what you meant. Here is one that came to mind (https://cuelang.org/play/?id=gOcQzuw6cxl#cue@export@json):
```
#Series: {
    name: string
    data: string
}

_ResponseTypes: [
    "BytesSent",
    "BytesReceived",
    "BytesPending"
]

#Response: {
    [or(_ResponseTypes)]: [...#Series]
}
```

Would something like this work for your use case?

**Reply by hooksie1** [reply]
Ah yeah that looks like it might work.  Thanks for the pointer. Still learning the language and I didn't realize the `[or` was a thing.
